std/string.d

```diff
@@ -3992,6 +3999,21 @@ auto detabber(Range)(Range r, size_t tabSize = 8)
     assert(detabber(" \n\tx", 9).array == " \n         x");
 }
 
+unittest
+{
+    import std.array : array;
+
+    static struct TestStruct
+    {
+        string s;
+        alias s this;
```

Comment:
The really bad part of https://issues.dlang.org/show_bug.cgi?id=15027 is that DirEntry does not alias a field that can be taken by reference, so maybe this should be used in the unittest, e.g.
```
   string name() { return s; }
   alias name this;
```